Sources/StructuredQueriesCore/SQLite/JSONFunctions.swift

```diff
@@ -34,3 +37,77 @@ extension QueryExpression where QueryValue: Codable & Sendable {
     AggregateFunction("json_group_array", self, order: order, filter: filter)
   }
 }
+
+extension PrimaryKeyedTableDefinition where QueryValue: Codable & Sendable {
+  /// A JSON array repsentation of the aggregation of a table's columns.
+  ///
+  /// Constructs a JSON array of JSON objects with a field for each column of the table. This can be
+  /// useful for loading many associated values in a single query. For example, to query for every
+  /// reminders list, along with the array of reminders it is associated with, one can define a
+  /// custom `@Selection` for that data and query as follows:
+  ///
+  /// @Row {
+  ///   @Column {
+  ///     ```swift
+  ///     @Selection struct Row {
+  ///       let remindersList: RemindersList
+  ///       @Column(as: JSONRepresentation<[Reminder]>.self)
+  ///       let reminders: Reminder
+  ///     }
+  ///     RemindersList
+  ///       .join(Reminder.all) { $0.id.eq($1.remindersListID) }
+  ///       .select {
+  ///         Row.Columns(
+  ///           remindersList: $0,
+  ///           reminders: $1.jsonObjects
+  ///         )
+  ///       }
+  ///     ```
+  ///   }
+  ///   @Column {
+  ///     ```sql
+  ///      SELECT
+  ///       "remindersLists".…,
+  ///       iif(
+  ///         "reminders"."id" IS NULL,
+  ///         NULL,
+  ///         json_object(
+  ///           'id', json_quote("id"),
+  ///           'title', json_quote("title"),
+  ///           'priority', json_quote("priority")
+  ///         )
+  ///       )
+  ///     FROM "remindersLists"
+  ///     JOIN "reminders"
+  ///       ON ("remindersLists"."id" = "reminders"."remindersListID")
+  ///     ```
+  ///   }
+  /// }
+  ///
+  /// > Note: If the primary key of the row is NULL, then the object is omitted from the array.
+  public var jsonObjects: some QueryExpression<JSONRepresentation<[QueryValue]>> {
```

Comment:
My only bike-shed here is to maybe overload `jsonGroupArray` here, since it's a specialized version.